pkg/kubectl/cmd/exec.go

```diff
@@ -174,6 +174,12 @@ func (p *ExecOptions) Run() error {
 		return err
 	}
 
+	started := pod.Status.StartTime != nil
+	terminated := pod.Status.Phase == api.PodSucceeded || pod.Status.Phase == api.PodFailed
+	if !started || terminated {
+		return fmt.Errorf("pod %s is not running and the command cannot be executed; current phase is %s", pod.Name, pod.Status.Phase)
```

Comment:
Might suggest communicating to the user directly - "cannot exec into a container in a completed pod (%s)".